--- regint.h.ori	2023-09-29 16:42:24.377227323 +0200
+++ regint.h	2023-09-29 23:15:50.075119315 +0200
@@ -61,7 +61,7 @@
 /* spec. config */
 #define USE_REGSET
 #define USE_CALL
-#define USE_CALLOUT
+// #define USE_CALLOUT
 #define USE_BACKREF_WITH_LEVEL        /* \k<name+n>, \k<name-n> */
 #define USE_WHOLE_OPTIONS
 #define USE_RIGID_CHECK_CAPTURES_IN_EMPTY_REPEAT        /* /(?:()|())*\2/ */
@@ -288,11 +288,7 @@
 #endif
 #endif /* _WIN32 */
 
-#if SIZEOF_VOIDP == SIZEOF_LONG
-typedef unsigned long hash_data_type;
-#elif SIZEOF_VOIDP == SIZEOF_LONG_LONG
-typedef unsigned long long hash_data_type;
-#endif
+typedef void* hash_data_type;
 
 /* strend hash */
 typedef void* hash_table_type;
